packages/bpk-theming/src/validateThemeAttributes.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { difference, omit } from 'lodash';
+
+const BpkThemeableText = [
+  'textColor',
+  'textBackgroundColor',
+];
+
+const themeAttributes = {
+  BpkThemeableText,
+};
+
+const validateThemeAttributes = (theme) => {
+  let validatedTheme = theme;
+  Object.keys(themeAttributes).forEach((componentName) => {
+    const attributesForComponent = themeAttributes[componentName];
+    const numOfAttributesMissingForComponent = difference(attributesForComponent, Object.keys(theme)).length;
```

Comment:
Doing `Object.keys(theme)` everytime is redundant and could be extracted before the `forEach`. Also worth considering whether we'd want to build a `Set` of the attributes since that would give us O(1) lookups and avoid using `difference` which is probably O(n^2).

Reply:
Good idea - I've extracted `Object.keys` out, will work on optimising the rest.

Reply:
Set isn't available in IE10 and Safari 10 which might be problematic. I'm not sure what our normal polyfills are configured for. cc @matthewdavidson

Reply:
I've removed the dependency on lodash in favour of using `Set.prototype.has` - what do you think? (Aside from the potential polyfill issue).

Reply:
Superb, but the polyfill issue is important so we should test this on IE10 and Safari 10

Reply:
So `Map` and `Set ` aren't included in our polyfills and thus aren't usable. We could build a object instead and get similar O(1) perfomance.